compass/milestone/MilestoneDatabase.java

```diff
@@ -76,18 +76,46 @@ public MilestoneDatabase(SpongeFactory.Mode powMode, SignatureSource signatureSo
     this.powMode = powMode;
   }
 
+  private static List<String> readInBatches(BufferedReader br, int batchSize, int totalSize) throws IOException {
+    List<String> result = new ArrayList<>(totalSize);
+    for (int i = 0; i < totalSize; i += batchSize) {
+      List<String> batch = readLinesInBatches(br, batchSize);
+      result.addAll(batch);
+      if (batch.size() < batchSize) {
+        return result;
+      }
+    }
+    return result;
+  }
+
+  private static List<String> readLinesInBatches(BufferedReader br, int batchSize) throws IOException {
```

Comment:
This method does not reduce memory consumption.
It only distracted me from the real solution on line 80...